lua/entities/kleiner/init.lua

```diff
@@ -0,0 +1,916 @@
+            end
+        end
+
+        self.IsAlive = false
+    end
+
+    local rag = self:BecomeRagdoll(dmginfo)
+end
+
+function ENT:BodyUpdate()
+    local act = ACT_IDLE
+
+    if (self.loco:GetVelocity():Length() > 1) then
+        act = ACT_WALK
+    end
+
+    if (self.loco:GetVelocity():Length() > 150) then
+        act = ACT_RUN
+    end
+
+    if (not self:OnGround()) then
+        act = ACT_JUMP
+    end
+
+    if (self.ClimbDir == 1) then
+        act = ACT_IDLE
+    end
+
+    if (self.ClimbDir == -1) then
+        act = ACT_IDLE
+    end
+
+    self.MainActivity = act
+
+    if (self:GetActivity() ~= self.MainActivity) then
+        self:StartActivity(self.MainActivity)
+    end
+
+    self:BodyMoveXY()
+end
+
+function ENT:PullGrenades()
+    if (IsValid(self)) then
+        self:AddGestureSequence(self:LookupSequence("startleclipboardgesture"))
+
+        timer.Simple(2.4, function()
+            if (IsValid(self)) then
+                self:AddGestureSequence(self:LookupSequence("blowclipboardgesture"))
+            end
+        end)
+
+        self.Suicidal = true
+        local pos, ang = self:GetBonePosition(self:LookupBone("ValveBiped.Bip01_L_Hand") or 0)
+        pos = pos + ang:Right() * 2
+        pos = pos + ang:Forward() * 4
+        local gren = self:SpawnGrenade()
+        gren:SetPos(pos)
+        gren:SetAngles(ang)
+        gren:SetParent(self, 10)
+        self.Grenade1 = gren
+        local pos, ang = self:GetBonePosition(self:LookupBone("ValveBiped.Bip01_R_Hand") or 0)
+        pos = pos + ang:Right() * 2
+        pos = pos + ang:Forward() * 4
+        ang:RotateAroundAxis(ang:Right(), 180)
+        local gren = self:SpawnGrenade()
+        gren:SetPos(pos)
+        gren:SetAngles(ang)
+        gren:SetParent(self, 11)
+        self.Grenade2 = gren
+
+        timer.Simple(1, function()
+            if (IsValid(self)) then
+                if (IsValid(self:GetTarget())) then
+                    self:SetTargetViolence(self:GetTarget(), math.floor((self:GetTargetViolence(self:GetTarget()) - 1) / 1.15))
+                end
+
+                if (IsValid(self.Grenade1) and IsValid(self.Grenade1:GetParent())) then
+                    self.Grenade1.DamageTriggered = true
+                    self.Grenade1:Fire("SetTimer", 3)
+                end
+
+                if (IsValid(self.Grenade2) and IsValid(self.Grenade2:GetParent())) then
+                    self.Grenade1.DamageTriggered = true
+                    self.Grenade2:Fire("SetTimer", 3)
+                end
+            end
+        end)
+    end
+end
+
+function ENT:SpawnGrenade()
+    local gren = ents.Create("npc_grenade_frag")
+    gren:Spawn()
+    gren:Activate()
+    gren:SetMoveType(MOVETYPE_NONE)
+    gren:PhysicsDestroy()
+    gren:SetCollisionGroup(COLLISION_GROUP_WORLD)
+    gren:SetOwner(self)
+    gren.Owner = self
+
+    return gren
+end
+
+function ENT:DropGrenades()
+    if (IsValid(self.Grenade1)) then
```

Comment:
You could do `for _,gren in ipairs({self.Grenade1,self.Grenade2}) do` (not mandatory, but that is good practice)